sim/battle.ts

```diff
@@ -2987,6 +2989,24 @@ export class Battle extends Dex.ModdedDex {
 		return false;
 	}
 
+	hint(hint: string, once?: boolean, sideid?: 'p1'|'p2') {
```

Comment:
Out of interest, why are you using a sideid here? I would have thought a Side would make more sense (and also simplify the code slightly, see Pokemon.getHeathInner).

Reply:
I agree, it would be simpler. I'll update it in a future commit.